src/System.IdentityModel.Tokens.Jwt/JwtPayload.cs

```diff
@@ -236,6 +281,11 @@ internal void AddFirstPriorityClaims(string issuer, string audience, DateTime? n
             // if could be the case that some of the claims above had an 'aud' claim;
             if (!string.IsNullOrEmpty(audience))
                 AddClaim(new Claim(JwtRegisteredClaimNames.Aud, audience, ClaimValueTypes.String));
+
+            AddClaims(audiences
+                .Where(aud => !aud.IsNullOrEmpty())
```

Comment:
Can we please avoid linq here as well?

Reply:
I double-checked this appears to be the last one